src/Provider/products.provider.js

```diff
@@ -0,0 +1,34 @@
+import React, { createContext, useReducer } from "react";
+import productsConstant from "./products.constant";
+import axiosInstance from "../helper/axios";
+import productsReducer, { initialState } from "./products.reducer";
+
+export const ProductContext = createContext({
+  initialState,
+});
+
+const ProductsProvider = ({ children }) => {
```

Comment:
Số ít thôi, k cần số nhiều đâu anh
```suggestion
const ProductProvider = ({ children }) => {
```